google/cloud/security/inventory/pipeline_builder.py

```diff
@@ -0,0 +1,277 @@
+        self.config = file_loader.read_and_parse_file(config_path)
+        self.flags = flags
+        self.api_map = api_map
+        self.dao_map = dao_map
+
+    def _find_runnable_pipelines(self, root):
+        """Initialize the pipeline builder.
+
+        Args:
+            root: PipelineNode representing the top-level starting point
+                of the pipeline dependency tree. The entire pipeline
+                dependency tree are tuple of children PipelineNodes
+                of this root.
+                Example:
+                root.resource_name = 'organizations'
+                root.enabled = True
+                root.parent = None
+                root.children = (pipeline_node1, pipeline_node2, ...)
+
+        Returns:
+            runnable_pipelines: List of the pipelines that will be run. The
+                order in the list represents the order they need to be run.
+                i.e. going top-down in the dependency tree.
+        """
+        # If child pipeline is true, then all parents will become true.
+        # Even if the parent(s) is(are) false.
+        # Manually traverse the parents since anytree walker api doesn't
+        # make sense.
+        for node in anytree.iterators.PostOrderIter(root):
+            if node.enabled:
+                while node.parent is not None:
+                    node.parent.enabled = node.enabled
+                    node = node.parent
+
+        LOGGER.debug('Dependency tree of the pipelines: %s',
+                     anytree.RenderTree(root, style=anytree.AsciiStyle())
+                     .by_attr('resource_name'))
+        LOGGER.debug('Which pipelines are enabled: %s',
+                     anytree.RenderTree(root, style=anytree.AsciiStyle())
+                     .by_attr('enabled'))
+
+        # Now, we have the true state of whether a pipeline should be run.
+        # Get a list of pipeline instances that will actually be run.
+        # The order matters: must go top-down in the tree, by PreOrder.
+        # http://anytree.readthedocs.io/en/latest/apidoc/anytree.iterators.html
+        runnable_pipelines = []
+        for node in anytree.iterators.PreOrderIter(root):
+            if node.enabled:
+                module_path = 'google.cloud.security.inventory.pipelines.{}'
+                module_name = module_path.format(
+                    self.REQUIREMENTS_MAP
+                    .get(node.resource_name)
+                    .get('module_name'))
+                try:
+                    module = importlib.import_module(module_name)
+                except (ImportError, TypeError, ValueError) as e:
+                    LOGGER.error('Unable to import %s\n%s', module_name, e)
+                    continue
+
+                class_name = (
+                    self.REQUIREMENTS_MAP
+                    .get(node.resource_name)
+                    .get('module_name')
+                    .title()
+                    .replace('_', ''))
```

Comment:
It's not clear to me what you're doing with class_name here and why we need to replace the underscores? Shouldn't they just match? A comment would be useful here.